### ISSUE TITLE
returning response headers from origins

### ISSUE THREAD

**Issue by EyePulp** [cshum/imagor#416]
We have a an nginx caching layer that sits in front Imagor.  It uses headers from Imagor (cache-control, expires, last-modified, etc) to decide how long to cache things at nginx.

The problem is we have a about 6 different http sources we point Imagor at, and the images there have  different cache needs, and reliably provide headers about those needs.

What would be ideal would be an Imagor server argument such as `HTTP_LOADER_RETURN_RESPONSE_HEADERS=cache-control,expires` that would support a case-insensitive csv list of header names that if they existed on the http source response would override conflicting values from imagor and be returned as the headers from Imagor as part of its response.

This lets us keep the fallback of `IMAGOR_CACHE_HEADER_TTL` and related arguments for default behavior, but control specific response headers more granularly, if desired.

I haven't looked deep enough into Imagor's architecture to attempt to implement it, but I wanted to check with other users and @cshum to see if there's any interest (or if there's a better way to approach my problem!)

Thanks.

**Comment by cshum** [comment]
It makes sense to allow proxying/overriding HTTP response header, but should be disabled by default due to the untrusted nature.

Suggesting the server argument be named `HTTP_LOADER_OVERRIDE_RESPONSE_HEADERS`, to make it clear its an override in case of overlapping values?

**Comment by EyePulp** [comment]
@cshum I agree it should be disabled by default (and Imagor will behave exactly the same as today).  It's purely an optional override for users who trust their origin http sources & headers.

I think `HTTP_LOADER_OVERRIDE_RESPONSE_HEADERS` is a good name.  It's a difficult concept to capture with only a few words, so we can use the docs to be explicit about how it works.  

Another name I thought of (but is kind of long) was `HTTP_LOADER_PREFERRED_SOURCE_RESPONSE_HEADERS`  -- that way it's showing what headers you _want_ or _desire_ to use _if available_, but doesn't guarantee that result  -- if the header is missing from the source, we fall back to normal `Imagor` behavior.

I think either name is good and support whichever one you choose (it's your project after all). 

Make your choice, and then let me know if you want to build it or do you want me to attempt it?

Also thanks for your time and attention!

**Comment by cshum** [comment]
Hi @EyePulp would you please test it under the docker build of `develop` branch?

https://github.com/cshum/imagor/pkgs/container/imagor/191350415?tag=develop

```
ghcr.io/cshum/imagor@sha256:92ce46156699b80587ca6c11bc3db507750994c5ebd8d08b1770bf5072ffee05
```
Argument
```
-http-loader-override-response-headers string
        Override HTTP Loader response header to image response by csv e.g. Cache-Control,Expires
```

**Comment by EyePulp** [comment]
Hey @cshum, thanks for the quick work!

I'll set up that test locally and let you know the results tonight or tomorrow.  Thanks again!

**Comment by EyePulp** [comment]
@cshum - I wasn't able to get it to work, but that might be a mistake in how I was running it.  Here's how I'm testing

Docker compose service setup:

```yaml
version: "3.7"

services:
  imagor:
    image: ghcr.io/cshum/imagor@sha256:49fbfad50571738c8ead73b6bc79541c4d7d430ec1670e883cc2370adbed19d7
    environment:
      PORT: 8000
      DEBUG: 1
      IMAGOR_UNSAFE: 1
    command: -http-loader-override-response-headers cache-control

    ports:
      - "8000:8000"
```

Then I curl the original image to get a response header
```sh
curl -I https://raw.githubusercontent.com/cshum/imagor/master/testdata/gopher.png

# HTTP/2 200 
# cache-control: max-age=300
```

now I curl though imagor:
```sh
curl -i localhost:/8000/unsafe/200x0/https://raw.githubusercontent.com/cshum/imagor/master/testdata/gopher.png

# HTTP/1.1 200 OK
# Cache-Control: public, s-maxage=604800, max-age=604800, no-transform, stale-while-revalidate=86400
```

Please let me know if I messed up the testing.  Thanks again for working on this.

**Comment by cshum** [comment]
@EyePulp try again with the new build?

```
version: "3.7"

services:
  imagor:
    image: ghcr.io/cshum/imagor@sha256:8d175514232900745734bb9bf8d0aa6d9a938042c351510ee4ccf1cbf63f732d
    environment:
      PORT: 8000
      DEBUG: 1
      IMAGOR_UNSAFE: 1
      HTTP_LOADER_OVERRIDE_RESPONSE_HEADERS: cache-control
    ports:
      - "8000:8000"
```

**Comment by EyePulp** [comment]
@cshum I tested both the environment var and command line argument of the new version, and both worked perfectly!  Nice job!

**Comment by cshum** [comment]
Available in v1.4.11

**Comment by EyePulp** [comment]
@cshum Amazing! thanks so much for implementing that.
